mplc/multi_partner_learning.py

```diff
@@ -5,42 +5,34 @@
 
 import operator
 import os
-import pickle
+from abc import ABC, abstractmethod
 from timeit import default_timer as timer
 
-import matplotlib.pyplot as plt
 import numpy as np
-from keras.backend.tensorflow_backend import clear_session
+from keras.backend import clear_session
 from keras.callbacks import EarlyStopping
 from loguru import logger
-from sklearn.externals.joblib import dump, load
-from sklearn.linear_model import LogisticRegression
-from sklearn.metrics import log_loss
 from sklearn.preprocessing import normalize
 
 from . import constants
+from .mpl_utils import History, UniformAggregator
+from .partner import PartnerMpl
 
 
-class MultiPartnerLearning:
-
+class MultiPartnerLearning(ABC):
     def __init__(self,
                  partners_list,
                  epoch_count,
                  minibatch_count,
                  dataset,
-                 multi_partner_learning_approach,
-                 aggregation_weighting="uniform",
+                 aggregation=UniformAggregator,
```

Comment:
Can we change the default to data_volume?

Reply:
Ok for me

Reply:
Done !